### PR TITLE
feat: events apps new install flow

### LINKED ISSUE #13414: [CAL-3029] 2024-05-23 - App store install flow
Take over from @morgans existing work and complete the following tasks

refactor EventTypeAppInterfaces of the apps used in event-types settings

- [X] split Form part from the APPcard part
- [X] already done for qr_code app ([commit here](https://github.com/calcom/cal.com/pull/11975/commits/ca0edf1ee7165b74195eb1da24ddbf408a84e12e))
- [X] check if there are new oAuth apps added, if yes add isOAuth to app_metadata.ts
- [X] adapt oauth apps api callback / add to redirect to new app install flow
- [X] When clicking on install new app button, only redirect to the new app install flow if it is an oAuth app or an app that extends the EventType feature in it's config
- [X] pr was done commit by commit for easier readability, also self review comments on the github pr

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3029](https://linear.app/calcom/issue/CAL-3029/app-store-install-flow)</sub>

### PR DESCRIPTION
**Pull request by ibex088**
## What does this PR do?

Fixes #13414
Fixes CAL-3029

[LOOM 1](https://www.loom.com/share/933c2663d7eb467e989425b75bc98314?sid=cdfdd879-517b-4e82-ac87-d482d52aba75)
[LOOM2](https://www.loom.com/share/35af734684ca4cc48f753ccd0b62e73c?sid=7c1f672b-c95d-40b0-9a0d-6ad49384bbc2)

this PR is a collective PR for different PRs(#11975 and #14077) that were tested individually and merged together into this pr 

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is a UI/UX design document, please, share it here.

## Type of change

- New feature (non-breaking change which adds functionality)
- Tests (Unit/Integration/E2E or any other test)

## How should this be tested?
checkout the looms, they explain the "How" to in great detail.

-testcasses:
* for all the apps that dont extend event-types there is no change.
* for all the apps that extend event-types i.e. all the analytics apps + payment apps + qr_code + Giphy + basecamp3 we follow the new flow
* for the apps that go to the new flow we can click on the skip button and enable apps individually in event-type/apps

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Review by joeauyeung — CHANGES_REQUESTED** [review]
Great work so far @SomayChauhan. I have on blocking comment which I think is causing some of these test cases to fail 

## Test Cases
- Conferencing apps
	- OAuth (Zoom)
		- From app store ❌ 
		- From app page ✅ 
	- Link based (Campfire) ✅
		- From app store ❌ 
		- From app page ✅ 
- Calendar apps (GCal) 
	- From app store ❌ 
	- From app page ✅ 
- Automation apps 
	- Basic (N8n) ✅ 
	- Setup page (Zaiper) ✅ 
- Analytics (GTM) ✅ 
	- From event type ✅
- Messaging (Telegram)  
	- From app store ❌ 
	- From app page ✅ 
- CRM (Zoho CRM) ✅ 
	- For now until https://github.com/calcom/cal.com/pull/14450 is merged
- Payment (Stripe) ✅ 
	- From event type ✅  

## Blocking
- From the main app store page. I'm not able to install apps that don't have the team selector dropdown

https://github.com/calcom/cal.com/assets/65426560/82598b2c-2647-4b30-8082-54f28994a430

## Non-blocking
- As a follow up, I think with the new flow we can vastly improve the performance of the app store page by removing the need for this drop down of the user and their teams and just redirect to the new app install flow.

**Comment by SomayChauhan** [comment]
Hats off @joeauyeung for spotting that, Pushed a fix!

**Comment by SomayChauhan** [comment]
>As a follow up, I think with the new flow we can vastly improve the performance of the app store page by removing the need for this drop down of the user and their teams and just redirect to the new app install flow.

I see what you mean, that'd be Great!!, I'll work on it.

**Review by joeauyeung — DISMISSED** [review]
Great fix @SomayChauhan. All failed test cases are passing. I'll approve for now but we should still have one more as this is labeled as high risk.

File: apps/api/v2/tsconfig.json
Comment (zomars) [inline apps/api/v2/tsconfig.json:1]: Is this change intended?

Comment (SomayChauhan) [reply apps/api/v2/tsconfig.json:1]: umm, maybe @ThyMinimalDev knows?

**Review by zomars — CHANGES_REQUESTED** [review]
Can we also provide some evidence in a loom? I would like to see a before and after. The PR description doesn't determine what should I be testing. Reading the code itself doesn't provide enough context.

**Comment by SomayChauhan** [comment]
@zomars Added the Looms to the pr description

**Comment by zomars** [comment]
> Can we also provide some evidence in a loom? I would like to see a before and after. The PR description doesn't determine what should I be testing. Reading the code itself doesn't provide enough context.

Thanks @SomayChauhan can we please fill the PR description template as well? 

https://raw.githubusercontent.com/calcom/cal.com/main/.github/PULL_REQUEST_TEMPLATE.md

Also it seems we need to update Stripe E2E tests. These are consistently failing:

image

**Comment by zomars** [comment]
@SomayChauhan please also add the "## How should this be tested?" part of the PR description. I think that's the most important one for reviewers.

**Comment by PeerRich** [comment]
as a follow up PR i would love to have this onboarding flow when adding a new conferencing tool to select which event type it should be installed on as a location

adding conferencing apps and then going to each event type manually is not intuitive right now

File: packages/ui/components/scrollable/ScrollableArea.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ const ScrollableArea = ({ children, className }: PropsWithChildren<{ className?:
         className // Pass in your max-w / max-h
       )}>
       {children}
-      {isOverflowingY && <div style={overflowIndicatorStyles} data-testid="overflow-indicator" />}
```
Comment (zomars) [inline packages/ui/components/scrollable/ScrollableArea.tsx:38]: Why was this removed? We get a warning of unused styles. If it's not needed then let's remove it completely.

**Review by zomars — COMMENTED** [review]
We're almost there. Just need a little bit more context. IDK if you or @ThyMinimalDev can provide it. Why was this reverted in the first place? Was was failing back then that should be fixed by now?

**Comment by zomars** [comment]
On another note. It seems like for some reason this PR is messing with rescheduling behavior? 
image

**Comment by SomayChauhan** [comment]
> We're almost there. Just need a little bit more context. IDK if you or @ThyMinimalDev can provide it. Why was this reverted in the first place? Was was failing back then that should be fixed by now?

The revert occurred because we were redirecting the installation of Zoom to the flow
https://app.campsite.co/cal/posts/jxcifntaflzj#comment-4qpbzzeicy5m
which was later fixed by https://github.com/calcom/cal.com/pull/14527. (This commit has been merged into the pull request).

**Comment by SomayChauhan** [comment]
> On another note. It seems like for some reason this PR is messing with rescheduling behavior?

looking into it

File: packages/ui/components/scrollable/ScrollableArea.tsx (hunk 1 above)
Comment (SomayChauhan) [reply packages/ui/components/scrollable/ScrollableArea.tsx:38]: Screenshot 2024-04-25 at 1 58 37 PM
it adds this shadow at the top, i think we can remove this

Comment (SomayChauhan) [reply packages/ui/components/scrollable/ScrollableArea.tsx:38]: removed 👍

**Comment by SomayChauhan** [comment]
not entirely sure if the failing tests are related

**Comment by zomars** [comment]
Found this while testing. owner1 is both owner of a team and an organization. It has the following event types:

image

I'm trying to install Stripe for the owner1 profile (not the team):

image

But when coming back from the Stripe callback I get presented with both the org and team event types to setup. Even tho I only installed it for the user.

image

 Is this the intended behavior? Shouldn't we only show the event types of the entity is being currently installed on?

**Comment by SomayChauhan** [comment]
> Is this the intended behavior? Shouldn't we only show the event types of the entity is being currently installed on?

this should definately not be the case, looking into this

**Comment by SomayChauhan** [comment]
@zomars pushed a fix!!

**Comment by SomayChauhan** [comment]
There was an unexpected merge [change](https://github.com/calcom/cal.com/pull/14616/commits/7b33dec291b001199b681a47c9f432759af4f73b) that likely slipped in during the process of merging and reverting branches, leading to test failures
all green now 👍

**Review by zomars — APPROVED** [review]
LEt's go!! Waiting on checks to pass

**Comment by keithwillcode** [comment]
@SomayChauhan Failing Stripe test. https://github.com/calcom/cal.com/actions/runs/9213084475/job/25346504639
